### PR TITLE
Vuefy Number and Slider Form inputs

### LINKED ISSUE #12874: [form parameter] integer, float: "_fieldSlider"
This is an issue for tracking/assigning conversion of the backbone form input (found here:  https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Form/Elements/parameters.js) into a vue component.

For more details see https://github.com/galaxyproject/galaxy/issues/11076

### PR DESCRIPTION
**Pull request by OlegZharkov**
For whatever reason, it's called `_fieldSlider`, even though it also handles regular input and floats. So I decided to rename it to FormNumber that IMHO better represents the component
resolve https://github.com/galaxyproject/galaxy/issues/12874
xref: https://github.com/galaxyproject/galaxy/issues/11076#issuecomment-958261281

Regular number form with and without range
image

Integer example:
Peek 2021-11-24 18-23
## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/Tool/ToolForm.vue [hunk 1]
```diff
@@ -224,6 +224,7 @@ export default {
         },
         onValidation(validationInternal) {
             this.validationInternal = validationInternal;
+            this.validationInternal = validationInternal;
```
Comment (guerler) [inline client/src/components/Tool/ToolForm.vue]: Duplicate.

Comment (dannon) [inline client/src/components/Tool/ToolForm.vue]:
```suggestion
```

Comment (OlegZharkov) [reply client/src/components/Tool/ToolForm.vue]: that's probably a merge artifact, thanks!

**Comment by OlegZharkov** [comment]
This depends on https://github.com/galaxyproject/galaxy/pull/12984 now

**Comment by guerler** [comment]
@OlegZharkov thanks for this, looks great. While looking at this I noticed a minor issue: 
image
Apparently it requires explicitly `42.0` instead of just `42`.
Also do you think it would work to highlight the whole input element as `info` and show the notification as text above for more consistency like here:
image
If you want to keep the error message separately I suggest to show it above the input element with less vertical padding and without the `x` mark to remove it since it disappears by itself.

**Comment by guerler** [comment]
I tried it again and the above error did not appear, the out of range warning has been modified too. Thank you @OlegZharkov

**Comment by OlegZharkov** [comment]
Lint error doesn't seem to be related, I tried to run it locally and got 

```
(.venv) oleg@pc:~/galaxy/client$ yarn run prettier-check
yarn run v1.22.17
$ prettier --check 'src/style/scss/**/*.scss' 'src/**/{*.js,*.vue}' '!src/libs/**'
Checking formatting...
All matched files use Prettier code style!
Done in 25.16s.
```

**Comment by dannon** [comment]
Thanks one last time @OlegZharkov! ;)

**Pull request merged**